adapter/pubsub/pubsub.cc

```diff
@@ -0,0 +1,192 @@
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *
+ * Distributed under BSD 3-Clause license.                                   *
+ * Copyright by The HDF Group.                                               *
+ * Copyright by the Illinois Institute of Technology.                        *
+ * All rights reserved.                                                      *
+ *                                                                           *
+ * This file is part of Hermes. The full Hermes copyright notice, including  *
+ * terms governing use, modification, and redistribution, is contained in    *
+ * the COPYING file, which can be found at the top directory. If you do not  *
+ * have access to the file, you may request a copy from help@hdfgroup.org.   *
+ * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+#include "pubsub.h"
+
+namespace hapi = hapi;
+
+hapi::Status hermes::pubsub::mpiInit(int argc, char **argv) {
+  LOG(INFO) << "Starting MPI" << std::endl;
+  int mpi_threads_provided;
+  MPI_Init_thread(&argc, &argv, MPI_THREAD_MULTIPLE, &mpi_threads_provided);
+  if (mpi_threads_provided < MPI_THREAD_MULTIPLE) {
+    fprintf(stderr, "Didn't receive appropriate MPI threading specification\n");
+    return hapi::Status(hermes::INVALID_FILE);
+  }
+  return hapi::Status(hermes::HERMES_SUCCESS);
+}
+
+hapi::Status hermes::pubsub::connect(const std::string &config_file) {
+  LOG(INFO) << "Connecting adapter" << std::endl;
+  auto mdm = hermes::adapter::Singleton
+      <hermes::adapter::pubsub::MetadataManager>::GetInstance();
+  try {
+    mdm->InitializeHermes(config_file.c_str());
+    if (mdm->isClient()) {
+      return hapi::Status(hermes::HERMES_SUCCESS);
+    } else {
+      return hapi::Status(hermes::INVALID_FILE);
+    }
+  } catch (const std::exception& e) {
+    LOG(FATAL) << "Could not connect to hermes daemon" <<std::endl;
+    return hapi::Status(hermes::INVALID_FILE);
+  }
+}
+
+hapi::Status hermes::pubsub::connect() {
```

Comment:
Rather than duplicating all the code from above, can this function just call the other `connect`?

```C++
hapi::Status hermes::pubsub::connect() {
  char* hermes_config = getenv(kHermesConf);
  return hermes::pubsub::connect(hermes_config);
}
```